rca/static_src/javascript/components/scholarship-list.js

```diff
@@ -0,0 +1,42 @@
+class ScholarshipList {
+    static selector() {
+        return '[data-scholarships-url]';
+    }
+
+    constructor() {
+        this.form = document.getElementById('scholarship-form');
+        this.dataURL = this.form.getAttribute('data-scholarships-url');
+        this.programmeChooser = document.getElementById('id_programme');
+
+        this.getData();
+        this.bindEvents();
+    }
+
+    getData() {
+        // Ideally this would be refactored to use fetch
+        const xmlhttp = new XMLHttpRequest();
+        xmlhttp.onreadystatechange = function getScholarships() {
+            if (this.readyState === 4 && this.status === 200) {
+                const dataLocation = document.getElementById('id_scholarships');
```

Comment:
Could keep this with the other vars defined in the constructor.

Reply:
Agreed, but it doesn't seem to work, confused me…